packages/eslint-plugin-wdio/lib/rules/await-expect.js

```diff
@@ -0,0 +1,37 @@
+module.exports = {
+    rules : {
+        'await-expect' : {
+            meta   : {
+                type : 'problem',
+                docs : {
+                    description : 'expect must be prefixed with await',
+                    category    : 'Possible Errors',
+                    url         : 'https://github.com/webdriverio/packages/eslint-plugin-wdio/docs/rules/await-expect.md',
+                    recommended : true,
+                },
+                messages : {
+                    missingAwait : 'Missing await before an expect statement'
+                },
+                hasSuggestions : true
+            },
+            create : function(context) {
+                return {
+                    CallExpression(node) {
+                        const name = node.callee.name
+
+                        if (name === 'expect') {
+                            const parentType = node.parent && node.parent.parent && node.parent.parent.parent ? node.parent.parent.parent.type : false
+
+                            if (parentType && parentType !== 'AwaitExpression') {
```

Comment:
I wonder if it makes sense to also check the expression that is being expected, if it isn't a WebdriverIO object there might not be a need for `await`, e.g.:
```js
const title = browser.getTitle();
expect(title).toBe('foobar');
```
Given it might be very difficult to determine that I suggest to only explicitly apply this rule for all assertions we've defined in expect-webdriverio, what do you think?

Reply:
That is impossible to say given that a linter doesn't run your code. If we only do this for expect-webdriverio assertions we can be assured this is the case though.

Reply:
I rarely use typescript and haven't set up a project with it before in a long time but I'm not sure I really see the benefit to using typescript with eslint. I could use optional chaining but that would be the only benefit I think of vs having the overhead of dealing with builds and typescript.

I can see if it's possible to only enforce the await if it's a Webdriver object.

Reply:
Yeah, determining if it is a Webdriver object would't be possible.

One issue I see with not awaiting things like .toBe that I've ran across.

```
const title = browser.getTitle();

// Both of the below work
await expect(title).toBe('foobar');
expect(await title).toBe('foobar');
```

Ideally people should be doing the await on the title either before or while inside expect but if they forget the test can pass when it should have failed.